docs/spec-files/encryption.md

```diff
@@ -44,6 +44,69 @@ For example, following the guidelines of [message types](#message-types), an enc
+```jsonc
+{
+  "@context": "https://www.w3.org/ns/did/v1",
+  "id": "did:example:123456789abcdefghi",
+  ...
+  "keyAgreement": [
+    // this method can be used to perform key agreement as did:...fghi
+    "did:example:123456789abcdefghi#keys-1",
+    // this method is *only* approved for key agreement usage, it will not
+    // be used for any other verification relationship, so its full description is
+    // embedded here rather than using only a reference
+    {
+      "id": "did:example:123#zC9ByQ8aJs8vrNXyDhPHHNNMSHPcaSgNpjjsBYpMMjsTdS",
+      "type": "X25519KeyAgreementKey2019", // external (property value)
+      "controller": "did:example:123",
+      "publicKeyBase58": "9hFgmPVfmBZwRvFEyniQDBkz9LmV7gDEqytWyGZLmDXE"
+    }
+  ],
+  ...
+}
+```
+
+The envelope packing process should set the `skid` header with value `did:example:123456789abcdefghi#keys-1` in the envelope's protected headers and fetch the underlying key to execute ECDH-1PU key derivation for content key wrapping.
+
+Assuming she also has Bob's DID document which happens to include the following `KeyAgreement` section:
+
+```jsonc
+{
+  "@context": "https://www.w3.org/ns/did/v1",
+  "id": "did:example:jklmnopqrstuvwxyz1",
+  ...
+  "keyAgreement": [
+    {
+      "id": "did:example:jklmnopqrstuvwxyz1#key-1",
+      "type": "X25519KeyAgreementKey2019", // external (property value)
+      "controller": "did:example:jklmnopqrstuvwxyz1",
+      "publicKeyBase58": "9hFgmPVfmBZwRvFEyniQDBkz9LmV7gDEqytWyGZLmDXE"
+    }
+  ],
+  ...
+}
+```
+
+There should be only 1 entry in the recipients of the envelope, representing Bob. The corresponding `kid` header for this recipient MUST have `did:example:jklmnopqrstuvwxyz1#key-1` as value. The packing process MUST extract the public key bytes found in `publicKeyBase58` of Bob's DID Doc `KeyAgreement[0]` to execute the ECDH-1PU key derivation for content key wrapping.
+
+When Bob receives the envelope, the unpacking process on his end MUST resolve the `skid` protected header value using Alice's DID doc's `KeyAgreement[0]` in order to extract her public key. In Alice's DID Doc example above, `KeyAgreement[0]` is a reference id, it MUST be resolved from the main `VerificationMethod[]` of Alice's DID document (not shown in the example).
+
+Once resolved, the unpacker will then execute ECDH-1PU key derivation using this key and Bob's own recipient key found in the envelope's `recipients[0]` to unwrap the content encryption key.
+
+## Protecting the `skid` header
+When the `skid` cannot be revealed in a plain-text JWE header (to avoid potentially leaking sender's key id), the `skid` MAY be encrypted for each recipient. In this case, instead of having a `skid` protected header in the envelope, each recipient MAY include an `encrypted_skid` header with a value based on the encryption of `skid` using ECDH-ES `Z` computation of the `epk` and the recipient's key as the encryption key.
```

Comment:
We should move this portion into the 1PU spec. Are you in contact with Neil or do you want me to kick that discussion off with him?

Reply:
I think this concept can more generally be applied to anyone using the ECDH-1PU algorithm and will raise it there as a consideration. In the mean time, it makes sense to leave this in here. If we do add the text there, we can just reference the text from there and remove a few sentences from this paragraph.